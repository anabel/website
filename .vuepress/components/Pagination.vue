<script>
export default {
  name: "pagination",
  props: {
    page: {
      type: Number
    },
    pageSize: {
      type: Number,
      default: 10
    },
    numberOfItems: {
      type: Number
    }
  },
  computed: {
    pageCount() {
      return Math.ceil(this.numberOfItems / this.pageSize);
    },
    hasPrevious() {
      return this.page - 1 > 0;
    },
    hasNext() {
      return this.page + 1 <= this.pageCount;
    },
    next() {
      return this.page + 1;
    },
    previous() {
      return this.page - 1;
    }
  },
  methods: {
    isPage(page) {
      return page == this.page;
    }
  }
};
</script>

<template>
  <ul class="flex-1 max-w-xl mx-auto leading-normal mt-10">
    <li>
      <span v-if="hasPrevious" class="link text">
        <router-link :to="{path: '/posts/' + this.previous}">Anterior</router-link>
      </span>
      <span v-else class="link text unactive">Anterior</span>
    </li>
    <li v-for="page in pageCount">
      <span v-if="isPage(page)" class="selected number">{{page}}</span>
      <span v-else class="link number">
        <router-link :to="{path: '/posts/' + page}">{{page}}</router-link>
      </span>
    </li>
    <li>
      <span v-if="hasNext" class="link">
        <router-link :to="{path: '/posts/' + this.next}">Siguiente</router-link>
      </span>
      <span v-else class="link unactive">Siguiente</span>
    </li>
  </ul>
</template>

<style lang="stylus" scoped>
@import '../theme/styles/theme.styl';

ul {
  font-size: 0;
}

.link {
  @apply text-gray-600 py-2 px-3 border border-gray-600;}

.selected {
  @apply btn py-2 px-3 border border-gray-600;}

.link.unactive {
  @apply text-gray-500 border-gray-500;}

li {
  @apply inline text-sm font-sans;}
</style>>