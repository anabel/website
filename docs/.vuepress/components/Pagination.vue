<template>
  <nav class="pagination">
    <span v-if="hasPrevious()" class="btn float-left">
      <router-link :to="{path: '/posts/' + previous}">Anterior</router-link>
    </span>
    <span v-else class="btn unactive float-left">Anterior</span>
    <span v-if="hasNext()" class="btn float-right">
      <router-link :to="{path: '/posts/' + next}">Siguiente</router-link>
    </span>
    <span v-else class="btn unactive float-right">Siguiente</span>
  </nav>
</template>
<script>
export default {
  props: {
    previous: Number,
    next: Number,
    hasPrevious: Function,
    hasNext: Function
  }
};
</script>