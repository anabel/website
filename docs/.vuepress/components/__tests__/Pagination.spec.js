import Pagination from "../Pagination.vue";
import { mount, RouterLinkStub } from "@vue/test-utils";

describe("If hasPrevious is false", () => {
  const wrapper = mount(Pagination, {
    stubs: {
      RouterLink: RouterLinkStub
    },
    propsData: {
      hasPrevious: () => {
        return false;
      },
      hasNext: () => {
        return false;
      }
    }
  });

  it("previous should be disabled", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});

describe("If hasPrevious is true", () => {
  const wrapper = mount(Pagination, {
    stubs: {
      RouterLink: RouterLinkStub
    },
    propsData: {
      hasPrevious: () => {
        return true;
      },
      hasNext: () => {
        return false;
      }
    }
  });

  it("previous should be `a` tag", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});

describe("If hasPrevious is true and previous is 1", () => {
  const wrapper = mount(Pagination, {
    stubs: {
      RouterLink: RouterLinkStub
    },
    propsData: {
      hasPrevious: () => {
        return true;
      },
      hasNext: () => {
        return false;
      },
      previous: 1
    }
  });

  const expectedTo = { path: "/posts/1" };

  console.log(wrapper.find(RouterLinkStub).props().to);

  it("previous button should be `a` tag and have expected path", () => {
    expect(wrapper.find(RouterLinkStub).props().to).toStrictEqual(expectedTo);
  });
});

describe("If hasNext is false", () => {
  const wrapper = mount(Pagination, {
    stubs: {
      RouterLink: RouterLinkStub
    },
    propsData: {
      hasPrevious: () => {
        return false;
      },
      hasNext: () => {
        return false;
      }
    }
  });

  it("next should be disabled", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});

describe("If hasNext is true", () => {
  const wrapper = mount(Pagination, {
    stubs: {
      RouterLink: RouterLinkStub
    },
    propsData: {
      hasPrevious: () => {
        return false;
      },
      hasNext: () => {
        return true;
      }
    }
  });

  it("previous should be `a` tag", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});

describe("If hasNext is true and next is 1", () => {
  const wrapper = mount(Pagination, {
    stubs: {
      RouterLink: RouterLinkStub
    },
    propsData: {
      hasPrevious: () => {
        return false;
      },
      hasNext: () => {
        return true;
      },
      next: 1
    }
  });

  const expectedTo = { path: "/posts/1" };

  it("next button should be `a` tag and have expected path", () => {
    expect(wrapper.find(RouterLinkStub).props().to).toStrictEqual(expectedTo);
  });
});
