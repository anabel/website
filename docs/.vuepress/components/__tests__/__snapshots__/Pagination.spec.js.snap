// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`If hasNext is false next should be disabled 1`] = `<nav class="pagination"><span class="btn unactive float-left">Anterior</span> <span class="btn unactive float-right">Siguiente</span></nav>`;

exports[`If hasNext is true previous should be \`a\` tag 1`] = `<nav class="pagination"><span class="btn unactive float-left">Anterior</span> <span class="btn float-right"><a>Siguiente</a></span></nav>`;

exports[`If hasPrevious is false previous should be disabled 1`] = `<nav class="pagination"><span class="btn unactive float-left">Anterior</span> <span class="btn unactive float-right">Siguiente</span></nav>`;

exports[`If hasPrevious is true previous should be \`a\` tag 1`] = `<nav class="pagination"><span class="btn float-left"><a>Anterior</a></span> <span class="btn unactive float-right">Siguiente</span></nav>`;
